hi-ml/src/health_ml/run_ml.py

```diff
@@ -274,81 +268,78 @@ def is_crossval_disabled_or_child_0(self) -> bool:
             return self.container.crossval_index == 0
         return True
 
-    def get_trainer_for_inference(self, checkpoint_path: Optional[Path] = None) -> Trainer:
-        # We run inference on a single device because distributed strategies such as DDP use DistributedSampler
-        # internally, which replicates some samples to make sure all devices have the same batch size in case of
-        # uneven inputs.
+    def set_trainer_for_inference(self) -> None:
+        """ Set the runner's trainer for inference: validation or test.
+        We run inference on a single device because distributed strategies such as DDP use DistributedSampler
+        internally, which replicates some samples to make sure all devices have the same batch size in case of
+        uneven inputs which biases the results."""
         mlflow_run_id = get_mlflow_run_id_from_previous_loggers(self.trainer)
         self.container.max_num_gpus = 1
-
-        if self.container.run_inference_only:
-            assert checkpoint_path is not None
-        else:
-            self.validate_model_weights()
-
-        trainer, _ = create_lightning_trainer(
+        self.trainer, _ = create_lightning_trainer(
             container=self.container,
-            resume_from_checkpoint=checkpoint_path,
             num_nodes=1,
             azureml_run_for_logging=self.azureml_run_for_logging,
```

Comment:
Do you know if we have a test that ensures that metrics for the 3 different stages (training, validation, inference) still get logged to the same run?

Reply:
I think so, `test_get_mlflow_run_id_from_previous_loggers` added in #627

Reply:
I meant more like a test for  `set_trainer_for_inference` specifically that checks that the trainer at each stage has the expected properties (mlflow_run_id and azureml_run_for_logging, num_nodes etc)

Reply:
Good point, will add a test